js-api-doc/importer.d.ts

```diff
@@ -309,32 +309,120 @@ export interface Importer<sync extends 'sync' | 'async' = 'sync' | 'async'> {
+ * This is used for type checking. Users should use the exposed
+ * {@link nodePackageImporter} and not instantiate their own.
  *
  * @category Importer
  */
 export type NodePackageImporter = {
   _NodePackageImporterBrand: unknown;
 };
+
 /**
- * The built-in Node Package Importer, which is a Package Importer that resolves
- * using the standards and conventions of the Node ecosystem. It enables a
- * `pkg:` URL scheme for usage with `@use` that directs an implementation to
- * resolve a URL within a dependency.
+ * The built-in Node.js package importer. This loads pkg: URLs from node_modules
+ * according to the standard Node.js resolution algorithm.
  *
- * A Node Package Importer is exposed as a constant that can be added to the
+ * A Node.js package importer is exposed as a constant that can be added to the
  * `importers` option.
  *
  * @example
- *
  *```js
  *const sass = require('sass');
- *sass.compile('style.scss', {
+ *sass.compileString('@use "pkg:@angular/material', {
  *  importers: [sass.nodePackageImporter]
  *});
  *```
+ *
+ * Package authors can control what is exposed to their users through their
+ * `package.json` manifest. The recommended method is to add a `sass`
+ * conditional export to `package.json`.
+ *
+ * @example
+ * ```json
+ * // node_modules/uicomponents/package.json
+ * {
+ *   "exports": {
+ *     ".": {
+ *       "sass": "./src/scss/index.scss",
+ *       "import": "./dist/js/index.mjs",
+ *       "default": "./dist/js/index.js"
+ *     }
+ *   }
+ * }
+ * ```
+ *
+ * This allows a package user to write `@use "pkg:uicomponents"` to load the file at
+ * `node_modules/uicomponents/src/scss/index.scss`.
+ *
+ * The Node.js package importer supports the variety of formats supported by
+ * Node.js [package entry points], allowing authors to expose multiple subpaths.
+ *
+ * [package entry points]: https://nodejs.org/api/packages.html#package-entry-points
+ *
+ * @example
+ * ```json
+ * // node_modules/uicomponents/package.json
+ * {
+ *   "exports": {
+ *     ".": {
+ *       "sass": "./src/scss/index.scss",
+ *     },
+ *     "./colors": {
+ *       "sass": "./src/scss/_colors.scss",
+ *     },
+ *     "./theme/*": {
+ *       "sass": "./src/scss/theme/*.scss",
+ *     },
+ *   }
+ * }
+ * ```
+ *
+ * This allows a package user to write:
+ *
+ * - `@use "pkg:uicomponents";` to import the root export.
+ * - `@use "pkg:uicomponents/colors";` to import the colors partial.
+ * - `@use "pkg:uicomponents/theme/purple";` to import a purple theme.
+ *
+ * Note that while library users can rely on the importer to resolve
+ * [partials](https://sass-lang.com/documentation/at-rules/use#partials), [index
+ * files](https://sass-lang.com/documentation/at-rules/use#index-files), and
+ * extensions, library authors must specify the entire file path.
+ *
+ * In addition to the `sass` condition, the `style` condition is also
+ * acceptable, but authors should avoid using the `default` condition for Sass
```

Comment:
I'd say "and Sass will match the `default` condition if it's a relevant file type but authors are discouraged from relying on this". The current phrasing makes it ambiguous about whether `default` works _in practice_ or not.